governance/program/src/state/proposal.rs

```diff
@@ -273,14 +383,18 @@ impl Proposal {
                     .checked_div(MintMaxVoteWeightSource::SUPPLY_FRACTION_BASE as u128)
                     .unwrap() as u64;
 
+                let deny_vote_weight = self.deny_vote_weight.unwrap_or(0);
+
+                let max_option_vote_weight =
+                    self.options.iter().map(|o| o.vote_weight).max().unwrap();
```

Comment:
It's bit tricky.
In a simple scenario when we only have `Yes` (positive) and `No` (negative) votes then `sum(Yes)+sum(No) <= max` is always `true`. However when we use `mint supply fraction` as `max` then it's possible to have `sum(Yes)+sum(No) >  max` and we have to coerce `max` to equal `sum(Yes)+sum(No)` because there might be more tokens in circulation then the defined fraction (`max`).
In order to generalise this to multi choice scenario when users can vote on N options we have to find the highest scoring positive option `max(sum(positive)) + sum(No)` and coerce it, if needed, to  be `<= max`